### PR TITLE
Add support for convict-format-with-validator

### PR DESCRIPTION
**Pull request by louis79719**
Checklist
- [✅] Use a meaningful title for the pull request. Include the name of the package modified.
- [✅] Test the change in your own code. (Compile and run.)
- [✅] Add or edit tests to reflect the change. (Run with `npm test`.)
- [✅] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [✅] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [✅] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [✅] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [✅] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [✅] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [✅] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [✅] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [✅] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/convict-format-with-validator/convict-format-with-validator-tests.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import convictValidators = require('convict-format-with-validator');
+
+// $ExpectType string | undefined
+convictValidators.email.name;
+// $ExpectType ((val: any) => void) | undefined
```
Comment (mattmoreira) [inline types/convict-format-with-validator/convict-format-with-validator-tests.ts:5]: This type isn't any, is the return value from the convict: 
https://github.com/mozilla/node-convict/blob/95f4ab398719d002a81b8acb21d2317b7bdbc8b6/packages/convict/src/main.js#L458-L703

File: types/convict-format-with-validator/convict-format-with-validator-tests.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+convictValidators.email.validate;
+// $ExpectType ((val: any) => any) | undefined
+convictValidators.email.coerce;
```
Comment (mattmoreira) [inline types/convict-format-with-validator/convict-format-with-validator-tests.ts:8]: This type isn't any, is the return value from the convict: 

https://github.com/mozilla/node-convict/blob/95f4ab398719d002a81b8acb21d2317b7bdbc8b6/packages/convict/src/main.js#L458-L703

File: types/convict-format-with-validator/convict-format-with-validator-tests.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import convictValidators = require('convict-format-with-validator');
+
+// $ExpectType string | undefined
+convictValidators.email.name;
+// $ExpectType ((val: any) => void) | undefined
+convictValidators.email.validate;
```
Comment (mattmoreira) [inline types/convict-format-with-validator/convict-format-with-validator-tests.ts:6]: Should be a string according to:
https://github.com/validatorjs/validator.js/blob/master/src/lib/isEmail.js

Comment (louis79719) [reply types/convict-format-with-validator/convict-format-with-validator-tests.ts:6]: It's the type definition from convict
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/convict/index.d.ts

Comment (mattmoreira) [reply types/convict-format-with-validator/convict-format-with-validator-tests.ts:6]: Ok, as we agreed, let's keep the same for interface compatibility.

File: types/convict-format-with-validator/convict-format-with-validator-tests.ts (hunk 2 above)
Comment (mattmoreira) [reply types/convict-format-with-validator/convict-format-with-validator-tests.ts:8]: As we agreed, let's keep the same for interface compatibility

**Review by mattmoreira — APPROVED** [review]

**Pull request merged**